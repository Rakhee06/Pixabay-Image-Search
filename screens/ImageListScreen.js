import React, { Component } from 'react';
import {
    View,
    FlatList,
    ActivityIndicator,
} from "react-native";
import { ListItem } from "react-native-elements";
import { PIXABAY_API_KEY } from 'react-native-dotenv';
import axios from "axios";
import { MonoText } from '../components/StyledText';

export default class ImageListScreen extends Component {
    static navigationOptions = {
        title: 'List',
    };

    state = {
        searchText: this.props.navigation.state.params.event,
        apiUrl: 'https://pixabay.com/api',
        apiKey: PIXABAY_API_KEY,
        images: [],
        loading: true,
        error: null
    };

    componentDidMount() {
        this.searchImages();
    }

    // Function to fetch images based on search props passed by the home screen
    searchImages = () => {
        this.setState({ loading: true});
        axios.get(`${this.state.apiUrl}/?key=${this.state.apiKey}&q=${this.state.searchText}&image_type=photo&safesearch=true`)
            .then(res => this.setState({
                images: res.data.hits,
                loading: false
            }))
            .catch(error => {
                this.setState({ error, loading: false });
            });
        if(this.state.error){
            return<MonoText>Image Not Found! Please provide a valid input</MonoText>
        }

        return this.state.images;
    };

    renderSeparator = () => {
        return (
            <View
                style={{
                    height: 1,
                    width: "86%",
                    backgroundColor: "#478692",
                    marginLeft: "14%"
                }}
            />
        );
    };

    render() {
        const { navigate } = this.props.navigation;
        if (this.state.loading) {
            return (
                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                    <ActivityIndicator
                        size="large"
                        color='#478692'
                    />
                </View>);
        }
        else {
            return (
                <View>
                    <FlatList
                        data={this.state.images}
                        renderItem={({ item }) => (
                            <ListItem
                                roundAvatar
                                title={<MonoText>{item.user}</MonoText>}
                                leftAvatar={{ source: {uri: item.previewURL},
                                    onPress: () => navigate('Details' , {
                                        name: `${item.user}`,
                                        tags: `${item.tags}`,
                                        likes: `${item.likes}`,
                                        imageURL: `${item.largeImageURL}`,
                                        views: `${item.views}`
                                    }) }}
                            />
                        )}
                        keyExtractor={(item, index) => index.toString()}
                        ItemSeparatorComponent={this.renderSeparator}
                    />
                </View>
            );
        }
    }
};