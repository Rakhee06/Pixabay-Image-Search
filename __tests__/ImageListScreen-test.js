import 'react-native';
import React from 'react';
import ImageListScreen from '../screens/ImageListScreen';
import renderer from 'react-test-renderer';

describe("Image List Screen", () => {
    it("should render image list screen correctly", () => {
        const navigationMock = { state: { params: {
                    event: 'search'
                }} };

        const tree = renderer.create(<ImageListScreen navigation={navigationMock}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});