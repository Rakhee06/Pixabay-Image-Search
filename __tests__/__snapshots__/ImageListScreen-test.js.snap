// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image List Screen should render image list screen correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#478692"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
